### ISSUE TITLE
Reject attempt to override named tuple field in subclass

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7198]
Pyright should report an error if someone attempts to redefine a named tuple field in a subclass.

```python
class Point(NamedTuple):
    x: int

class BadPointWithName(Point):
    name: str = ""  # OK
    x: int = 0  # Should be a type error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.350, which I just published.

**Comment by fergusq** [comment]
@erictraut Should this apply to named tuple methods as well? We noticed that our type checking fails after the latest version of pyright, since we override methods in our NamedTuple classes. We have never noticed any problem with this, and the overriden methods work as expected. Is there a reason for the type checker to reject overriden methods?

Here is an example:

```py
from typing import NamedTuple

class a(NamedTuple):
    x = 0
    def f(self):
        print(self.x)

class b(a):
    def f(self):
        print(self.x + 1)

a().f()  # prints 0
b().f()  # prints 1
```

The above code works, yet pyright gives the following error:

```
/tmp/test.py
  /tmp/test.py:9:5 - error: Cannot override "f" because parent class "a" is a named tuple (reportIncompatibleVariableOverride)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
@fergusq, I agree that this shouldn't apply to methods. Apologies for the regression. Could you please file a new bug so this doesn't get lost?
